.github/workflows/build.yml

```diff
@@ -57,35 +64,59 @@ jobs:
         name: mac-binary
         path: build/asciidoctor-web-pdf-mac.zip
       if: matrix.primary
-    - name: Smoke test
-      run: |
-        unzip asciidoctor-web-pdf-linux.zip -d out
-        ./out/asciidoctor-web-pdf ./out/examples/document/basic-example.adoc -o basic.pdf
-        ls basic.pdf
-      working-directory: build
+    - name: Upload Linux artifact
+      uses: actions/upload-artifact@v2
+      with:
+        name: linux-binary
+        path: build/asciidoctor-web-pdf-linux.zip
       if: matrix.primary
+  test-linux:
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/download-artifact@v2
+        with:
+          name: linux-binary
+      - name: unzip
+        run: |
+          unzip asciidoctor-web-pdf-linux.zip -d out
+      - name: Smoke test
+        run: |
+          cd out/asciidoctor-web-pdf-linux-${{ needs.build.outputs.version }}
+          rm -rf basic.pdf
+          ./asciidoctor-web-pdf ./examples/document/basic-example.adoc -o basic.pdf
+          ls basic.pdf
+          test -f basic.pdf -a -s basic.pdf
   test-macos:
     needs: build
     runs-on: macos-latest
     steps:
       - uses: actions/download-artifact@v2
         with:
           name: mac-binary
-      - name: Smoke test
+      - name: unzip
         run: |
           unzip asciidoctor-web-pdf-mac.zip -d out
-          ./out/asciidoctor-web-pdf ./out/examples/document/basic-example.adoc -o basic.pdf
+      - name: Smoke test
+        run: |
+          cd out/asciidoctor-web-pdf-mac-${{ needs.build.outputs.version }}
+          rm -rf basic.pdf
+          ./asciidoctor-web-pdf ./examples/document/basic-example.adoc -o basic.pdf
           ls basic.pdf
+          test -f basic.pdf -a -s basic.pdf
   test-win:
     needs: build
     runs-on: windows-latest
     steps:
       - uses: actions/download-artifact@v2
         with:
           name: win-binary
-      - name: Smoke test
+      - name: unzip
         run: |
           unzip asciidoctor-web-pdf-win.zip -d out
-          out\asciidoctor-web-pdf.exe out\examples\document\basic-example.adoc -o basic.pdf
+      - name: Smoke test
+        run: |
+          cd out\asciidoctor-web-pdf-win-${{ needs.build.outputs.version }}
+          asciidoctor-web-pdf.exe examples\document\basic-example.adoc -o basic.pdf
```

Comment:
`test` is not available on Windows?

Reply:
I guess it is "exits" ..don't have a Windows machine around for testing. It would be good to test Github Actions locally. Found this project https://github.com/nektos/act, however Windows is not supported by now :-(